### ISSUE TITLE
Model refactoring: keras.Layer vs. function

### ISSUE THREAD

**Issue by b-nils** [alexander-pv/maskrcnn_tf2#15]
I have recently switched to this fork due to its modular design and backbone support. Currently, I am trying to understand the refactoring steps that were made...

Is there a reason why e.g. the function `rpn_graph()` ([source](https://github.com/matterport/Mask_RCNN/blob/3deaec5d902d16e1daf56b62d5971d428dc920bc/mrcnn/model.py#L830)) has been refactored to a dedicated class inheriting from `keras.Layer` ([source](https://github.com/alexander-pv/maskrcnn_tf2/blob/3909e5b607dc88754287a8f90a262e226a74ad64/src/layers/mrcnn_layers.py#L147)) and e.g. `fpn_classifier_graph()` ([source](https://github.com/alexander-pv/maskrcnn_tf2/blob/3909e5b607dc88754287a8f90a262e226a74ad64/src/layers/mrcnn_layers.py#L1124)) remains a function?

**Comment by alexander-pv** [comment]
Hi, @b-nils,

There are `FPNClassifier` and `FPNMaskLayer` in the subclassed version of the model which are classes inheriting from `tf.keras.layers.Layer`.
AFAIR, the reason is that fpn classifier and fpn mask include roi align layer. I did not take it out from them yet and that's why the model with functional API I mostly needed still has `fpn_classifier_graph` and `fpn_mask_graph`.
Perhaps, it's a good step forward more modular solution.

**Comment by b-nils** [comment]
Hey @alexander-pv,

thanks for the fast answer! I see, I have not looked into the sub-classed model version yet. Two follow-up questions:

1. Why are you still using `FMaskRCNN` per default? The sub-classed version is very modular and nicely structured... Is there no way saving it as `HDF5`?  - `FMaskRCNN` also uses some classes inheriting from `tf.keras.layers.Layer`. So what is the bottleneck? Just the `tf.keras.Model` inheritance?

1. Additionally I have seen that you are using the sub-classed `tf.keras.losses.Loss` classes independent from the chosen model (`FMaskRCNN` + `SMaskRCNN`). Have you noticed any performance loss or gain in terms of training speed compared to the functional `tf.keras` API?
